### PR TITLE
[11.11] Add support for additional parameters in `Projects::forks`

### PR DESCRIPTION
**Pull request by fjgarlin**
The `forks` endpoint allows a handy `search` parameter, so define it as an option. 

See the options for forks endpoint: https://docs.gitlab.com/ee/api/projects.html#list-forks-of-a-project

With the following code:
```
      $forks = $gitlab_client->projects()->forks($project, [
        'search' => 'search_term',
      ]);
```

**Currently**: throws exception as parameter is not defined.
**After applying changes**: returns list of forks that match "search_term"

### CODE REVIEW

File: src/Api/Projects.php [hunk 1]
```diff
@@ -1035,13 +1035,17 @@ public function languages($project_id)
 
     /**
      * @param int|string $project_id
-     * @param array      $parameters
+     * @param array      $parameters {
+     *
+     *     @var string $search         Return list of forks matching the search criteria (optional)
```
Comment (GrahamCampbell) [inline src/Api/Projects.php]: If we are going to document this, we need to document all the available parameters.

File: src/Api/Projects.php [hunk 2]
```diff
@@ -1035,13 +1035,17 @@ public function languages($project_id)
...
+     * }
      *
      * @return mixed
      */
     public function forks($project_id, array $parameters = [])
     {
         $resolver = $this->createOptionsResolver();
+        $resolver->setDefined('search');
```
Comment (GrahamCampbell) [inline src/Api/Projects.php:1088]: Shouldn't this be defined in the options resolver function?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Thanks for the PR. Please address the inline comments and add test coverage.

File: src/Api/Projects.php (hunk 2 above)
Comment (fjgarlin) [reply src/Api/Projects.php:1088]: I'm not too familiar with the code inside the library, but the rest of the methods seem to be defining the options in the same way, within the method.

File: src/Api/Projects.php (hunk 1 above)
Comment (fjgarlin) [reply src/Api/Projects.php]: I can try that. I wanted to get the minimum PR that would work and then have it reviewed by maintainers (aka you) to be sure that this was the right approach. I'll try to add and document the other options.

File: src/Api/Projects.php (hunk 2 above)
Comment (fjgarlin) [reply src/Api/Projects.php:1088]: I've done it now based on the current code (I took the `all` method as reference) and the information found here: https://docs.gitlab.com/ee/api/projects.html#list-forks-of-a-project

File: src/Api/Projects.php (hunk 1 above)
Comment (fjgarlin) [reply src/Api/Projects.php]: I've done it now based on the current code (I took the `all` method as a reference) and the information I found here: https://docs.gitlab.com/ee/api/projects.html#list-forks-of-a-project

File: src/Api/Projects.php (hunk 2 above)
Comment (fjgarlin) [reply src/Api/Projects.php:1088]: I added all the other options available.

**Comment by fjgarlin** [comment]
Inline comments were addressed and test coverage was added. Pleae review.

File: src/Api/Projects.php (hunk 2 above)
Comment (GrahamCampbell) [reply src/Api/Projects.php:1088]: Yeh, this is not correct. Please look over the `createOptionsResolver` method in this class, and only add the things not already in there, or if search is always applicable, instead, add it there.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Api/Projects.php (hunk 2 above)
Comment (fjgarlin) [reply src/Api/Projects.php:1088]: I don't think that `search` should bubble up to the `createOptionsResolver` as that's used by many other methods. Note that there is no override for `createOptionsResolver` in this class, it is using the one of `AbstractApi.php` file.

The line `$resolver->setDefined('search');` was taken from the `all` method within this same class.

Not sure what more changes are required? I might need some more details.

**Comment by GrahamCampbell** [comment]
Thanks.

**Pull request merged**